controllers/cluster/cluster_controller.go

```diff
@@ -153,7 +164,10 @@ func (r *ClusterReconciler) startProber(ctx context.Context, logger logr.Logger,
 // SetupWithManager sets up the controller with the Manager.
 func (r *ClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
-		For(&gardenerv1alpha1.Cluster{}).
+		For(
+			&gardenerv1alpha1.Cluster{},
+			ctrlbuilder.WithPredicates(workerLessShoot(mgr.GetLogger())),
```

Comment:
`mgr.GetLogger()` will not contain the controller name and kind. We should build the controller and use `c.GetLogger()` which is built with this info.
Ref: https://github.com/gardener/gardener/blob/b89462e43c1f56843d8d95bb1f459b1e0c0663b4/vendor/sigs.k8s.io/controller-runtime/pkg/builder/controller.go#L327-L336

Reply:
done